File: playwright/PageObjects/LoginPage.js
module.exports = class LoginPage {
  constructor(page) {
    this.page = page;
  }

  async enterEmail(email) {
    return this.page.type('input[name="identification"]', email);
  }

  async enterPassword(password) {
    return this.page.type('input[name="password"]', password);
  }

  async clickLogin() {
    return this.page.click('css=button.login');
  }
};

File: kraken/features/web/step_definitions/page_objects/login_page.rb
class LoginPage
  attr_reader :driver

  def initialize(driver)
    @driver = driver
  end

  def enter_email(email)
    @driver.find_element(
      :css, 'input[name="identification"]'
    ).send_keys(email)
  end

  def enter_password(password)
    @driver.find_element(
      :css, 'input[name="password"]'
    ).send_keys(password)
  end

  def click_login
    @driver.find_element(
      :css, 'button.login'
    ).click
  end

  def error_message
    @driver.find_element(
      :css, '.main-error'
    ).text
  end

  def clear_password
    @driver.find_element(
      :css, 'input[name="password"]'
    ).clear
  end
end
